src/lib.rs

```diff
@@ -572,91 +570,37 @@ impl Default for Ini {
     }
 }
 
-/// An iterator over the sections of an ini documet
+/// An iterator over the sections of an ini document
 pub struct IniIter<'a> {
     #[doc(hidden)]
     iter: ordered_hashmap::Iter<'a, String, Section>,
 }
 
 impl<'a> Iterator for IniIter<'a> {
-    type Item = (&'a String, SectionIter<'a>);
+    type Item = (&'a String, &'a Section);
 
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
-        self.iter.next().map(|(name, section)| (name, SectionIter { document: &section, iter: section.iter() }))
+        self.iter.next()
     }
 }
 
-/// A mutable iterator over the sections of an ini documet
+/// A mutable iterator over the sections of an ini document
 pub struct IniIterMut<'a> {
     #[doc(hidden)]
     iter: ordered_hashmap::IterMut<'a, String, Section>,
 }
 
 impl<'a> Iterator for IniIterMut<'a> {
-    type Item = (&'a String, SectionIterMut<'a>);
+    type Item = (&'a String, &'a mut Section);
 
     #[inline]
-    fn next(&mut self) -> Option<Self::Item> {
-        self.iter.next().map(|(name, section)| (name, SectionIterMut { iter: section.iter_mut() }))
-    }
-}
-
-type Section = OrderedHashMap<String, String>;
-
-/// An iterator over the entries of a section
-pub struct SectionIter<'a> {
-    #[doc(hidden)]
-    document: &'a Section,
-    iter: ordered_hashmap::Iter<'a, String, String>,
-}
-
-impl<'a> Iterator for SectionIter<'a> {
-    type Item = (&'a String, &'a String);
-
     fn next(&mut self) -> Option<Self::Item> {
         self.iter.next()
     }
 }
 
-impl<'a> SectionIter<'a> {
-    /// Get scalar value of key
-    ///
-    /// - output type `T` must implement [FromStr] trait for auto conversion
-    ///
-    /// # Example
-    /// ```
-    /// # use tini::Ini;
-    /// let conf = Ini::from_string("[section]\nkey=1\nvalue=2").unwrap();
-    ///
-    /// for (name, section) in conf.iter() {
-    ///     let key = section.get("key");
-    ///     let value = section.get("value");
-    ///     assert_eq!(key, Some(1));
-    ///     assert_eq!(value, Some(2));
-    /// }
-    /// ```
-    pub fn get<T>(&'a self, key: &str) -> Option<T>
-    where
-        T: FromStr,
-    {
-        self.document.get(key).and_then(|x| x.parse().ok())
```

Comment:
One API limitation the new version currently has - because it's just re-using the `OrderedHashMap` implementations, it does no parsing.

I expect that might need to be fixed.